src/jit/codegenxarch.cpp

```diff
@@ -6884,6 +6993,171 @@ void        CodeGen::genCompareLong(GenTreePtr  treeNode)
+//                      cmp loOp1,loOp2
+//                      ja  trueLabel
+//                      labelFinal:
+//
+//    GT_GE; unsigned   cmp hiOp1,hiOp2
+//                      ja  trueLabel
+//                      jb  labelFinal
+//                      cmp loOp1,loOp2
+//                      jae trueLabel
+//                      labelFinal:
+//
+//    GT_LT; signed     cmp hiOp1,hiOp
+//                      jl  trueLabel
+//                      jg  labelFinal
+//                      cmp loOp1,loOp2
+//                      jb  trueLabel
+//                      labelFinal:
+//
+//    GT_LE; signed     cmp hiOp1,hiOp2
+//                      jl  trueLabel
+//                      jg  labelFinal
+//                      cmp loOp1,loOp2
+//                      jbe trueLabel
+//                      labelFinal:
+//    
+//    GT_GT; signed     cmp hiOp1,hiOp2
+//                      jg  trueLabel
+//                      jl  labelFinal
+//                      cmp loOp1,loOp2
+//                      ja  trueLabel
+//                      labelFinal:
+//
+//    GT_GE; signed     cmp hiOp1,hiOp2
+//                      jg  trueLabel
+//                      jl  labelFinal
+//                      cmp loOp1,loOp2
+//                      jae trueLabel
+//                      labelFinal:
+//
+//    GT_EQ; signed     cmp hiOp1,hiOp2
+//                      jne labelFinal
+//                      cmp loOp1,loOp2
+//                      je  trueLabel
+//                      labelFinal:
+//
+//    GT_NE; signed     cmp hiOp1,hiOp2
+//                      jne labelTrue
+//                      cmp loOp1,loOp2
+//                      jne trueLabel
+//                      labelFinal:
+//
+// TODO-X86-CQ: Check if hi or lo parts of op2 are 0 and change the compare to a test.
+void        CodeGen::genJTrueLong(GenTreePtr  treeNode)
+{
+    assert(treeNode->OperIsCompare());
+
+    GenTreeOp *tree = treeNode->AsOp();
+    GenTreePtr op1 = tree->gtOp1;
+    GenTreePtr op2 = tree->gtOp2;
+
+    genConsumeOperands(tree);
+
+    assert(varTypeIsLong(op1->TypeGet()) && varTypeIsLong(op2->TypeGet()));
+    regNumber targetReg  = treeNode->gtRegNum;
+
+    GenTreePtr loOp1 = op1->gtGetOp1();
+    GenTreePtr hiOp1 = op1->gtGetOp2();
+    GenTreePtr loOp2 = op2->gtGetOp1();
+    GenTreePtr hiOp2 = op2->gtGetOp2();
+
+    // Create compare for the high parts
+    instruction ins = INS_cmp;
+    var_types cmpType = TYP_INT;
+    emitAttr cmpAttr = emitTypeSize(cmpType);
+
+    // Emit the compare instruction
+    getEmitter()->emitInsBinary(ins, cmpAttr, hiOp1, hiOp2);
+
+    // Generate the first jump for the high compare
+    CompareKind compareKind = ((tree->gtFlags & GTF_UNSIGNED) != 0) ? CK_UNSIGNED : CK_SIGNED;
+    
+    // TODO-X86-CQ: When targetReg == REG_NA, if the next block is a BBJ_ALWAYS, we can set
+    //              labelFinal = compiler->compCurBB->bbNext->bbJumpDest.
+    BasicBlock* labelHi = genCreateTempLabel();
```

Comment:
Isn't targetReg guaranteed to be REG_NA in this function? (If so, we should assert that.) In that case, why not just implement this and not leave it a TODO for later? If you don't want to implement it now, please add a GH issue.